### PR TITLE
feat: add Pareto goodness-of-fit statistics 

### LINKED ISSUE #53: Pareto distribution GoF criteria
All statistics are located in _pysatl_criterion/statistics_

To start, you can implement criteria that are applicable to different types of distributions, since they use the empirical distribution function and compare it with a reference distribution function. Such statistics are located in _pysatl_criterion/statistics/common.py_.

To implement a statistic for a specific distribution, you need to inherit from the appropriate base class (for example, for the Kolmogorov–Smirnov statistic for the exponential distribution, you can look at the class _**KolmogorovSmirnovExponentialityGofStatistic**_, which inherits from _**KSStatistic**_ and overrides the code and **execute_statistic** **methods**).

It is necessary to create a base statistic class for the given distribution, which will be inherited by all statistics of this distribution (for example, **_AbstractExponentialityGofStatistic_**).

The **code()** method should return a unique identifier of the criterion.

For each test statistic, it is necessary to calculate the value of this statistic manually or using other packages, and implement unit tests.

For each test, you need to add a docstring that specifies the parameters accepted by the method, the returned result, a brief description of how the statistic is calculated, and a reference to the article describing this statistic.

**Gitflow:**
Create your branch from _main_ branch name it **criteria/pareto**

**Plan:**

1. Add pysatl_criterion/statistics/pareto.py
2. Add abstract class **_AbstractParetoGofStatistic_**
3. Implement common criteria (like Kolmogorov-Smirnov, Anderson-Darling etc)
4. Inspect scientific articles for other Goodness-of-Fit test statistics
5. Implement Goodness-of-Fit test statistics
6. Verify Goodness-of-Fit test statistics writing unit tests
7. Create pull request to _main_ branch

It is **important** that all pipelines pass successfully.

### PR DESCRIPTION
**Pull request by prcmium**
## Summary
Add goodness-of-fit statistics for the **Pareto distribution** 
Solve the issue: #53 

- **Base Architecture:** Implemented `AbstractParetoGofStatistic` for Pareto-specific statistics.
- **Classic EDF Statistics:** Added support for Kolmogorov-Smirnov (`KS`), Anderson-Darling (`AD`), Cramér–von Mises (`CVM`), parameter-adaptive `Lilliefors`, and `Min Toshiyuki`.
- **Specialized Statistics:**
  - `Obradovic` (U-statistic based on characterization).
  - `Greenwood` (spacing-based test).
  - `Lequesne` (entropy & KL divergence test).
- **Testing:** Added unit tests covering logic, edge cases, and parameter validation for all Pareto GoF criteria.

## How to Test
Run pytest for the Pareto statistics test suite:
```bash
poetry run pytest

### CODE REVIEW

File: src/pysatl_criterion/distribution/distribution_type.py [hunk 1]
```diff
@@ -53,6 +53,9 @@ class DistributionType(Enum):
     SCALE_CON_NORMAL = "scale_con_normal"
     TRUNC_NORMAL = "trunc_normal"
     TUKEY = "tukey"
+    FISHER = "fisher"
+    RAYLEIGH = "rayleigh"
```
Comment (alex98247) [inline src/pysatl_criterion/distribution/distribution_type.py:57]: а зачем тут FISHER и RAYLEIGH?

File: src/pysatl_criterion/statistics/goodness_of_fit/pareto.py
Comment (f1i3g3) [inline src/pysatl_criterion/statistics/goodness_of_fit/pareto.py:1]: Возможно, можно ещё убрать часть функциональности в отдельные функции.

File: tests/distributions/test_pareto.py
Comment (f1i3g3) [inline tests/distributions/test_pareto.py:1]: rng = np.random.default_rng(42)
sample = scipy_stats.pareto.rvs(..., random_state=rng)
И это обернуть в фикстуру.

**Review by f1i3g3 — APPROVED** [review]
В целом, пойдет. Хотя можно попробовать улучшить.
Математику оставляю за @alex98247.

**Review by alex98247 — APPROVED** [review]
